shell/src/main/java/alluxio/shell/command/StatCommand.java

```diff
@@ -43,29 +49,101 @@ public String getCommandName() {
   }
 
   @Override
+  protected Options getOptions() {
+    return new Options().addOption(
+            Option.builder("f")
+                    .required(false)
+                    .hasArg()
+                    .desc("format")
+                    .build()
+    );
+  }
+
+  @Override
   protected void runCommand(AlluxioURI path, CommandLine cl) throws AlluxioException, IOException {
     URIStatus status = mFileSystem.getStatus(path);
-    if (status.isFolder()) {
-      System.out.println(path + " is a directory path.");
-      System.out.println(status);
+    if (cl.hasOption('f')) {
+      System.out.println(formatOutput(cl, status));
     } else {
-      System.out.println(path + " is a file path.");
-      System.out.println(status);
-      System.out.println("Containing the following blocks: ");
-      AlluxioBlockStore blockStore = AlluxioBlockStore.create();
-      for (long blockId : status.getBlockIds()) {
-        System.out.println(blockStore.getInfo(blockId));
+      if (status.isFolder()) {
+        System.out.println(path + " is a directory path.");
+        System.out.println(status);
+      } else {
+        System.out.println(path + " is a file path.");
+        System.out.println(status);
+        System.out.println("Containing the following blocks: ");
+        AlluxioBlockStore blockStore = AlluxioBlockStore.create();
+        for (long blockId : status.getBlockIds()) {
+          System.out.println(blockStore.getInfo(blockId));
+        }
       }
     }
   }
 
   @Override
   public String getUsage() {
-    return "stat <path>";
+    return "stat [-f <format>] <path>";
   }
 
   @Override
   public String getDescription() {
-    return "Displays info for the specified path both file and directory.";
+    return "Displays info for the specified path both file and directory."
+        + " Specify -f to display info in given format:"
+        + "   \"%N\": name of the file;"
+        + "   \"%z\": size of file in bytes;"
+        + "   \"%u\": owner;"
+        + "   \"%g\": group name of owner;"
+        + "   \"%y\" or \"%Y\": modification time,"
+            + " %y shows 'yyyy-MM-dd HH:mm:ss' (the UTC date),"
+            + " %Y it shows milliseconds since January 1, 1970 UTC;"
+        + "   \"%b\": Number of blocks allocated for file";
+  }
+
+  private static final String FORMAT_REGEX = "%([bguyzNY])";
+  private static final Pattern FORMAT_PATTERN = Pattern.compile(FORMAT_REGEX);
+
+  private String formatOutput(CommandLine cl, URIStatus status) {
+    String format = cl.getOptionValue('f');
+    int formatLen = format.length();
+
+    StringBuilder output = new StringBuilder();
+    Matcher m = FORMAT_PATTERN.matcher(format);
+    for (int i = 0; i < formatLen;) {
```

Comment:
for-loop like this without incrementing `i` explicitly is tricky to reason its correctness. 
Can you convert this into something like this (plz double check if my code is correct)

```java
int i = 0;
while (m.find()) {
  output.append(format.substring(i, m.start()));
  output.append(getField(m, status));
  i = m.end();
}
output.append(format.substring(i));
```